### PR TITLE
Added workflow with index and alignment

### PR DESCRIPTION
**Pull request by praveenraj2018**
Just add a sample wokflow that does the genome prep and star-alignment. It reads the sample FASTQ files from the sample sheet.

### CODE REVIEW

File: assets/samplesheet.valid.csv [hunk 1]
```diff
@@ -0,0 +1,2 @@
+sample,single_end,fastq_1,fastq_2,strandedness
```
Comment (maxulysse) [inline assets/samplesheet.valid.csv:1]:
```suggestion
sample,fastq_1,fastq_2,strandedness
```
I think that we can assess if it's single or pair end given the presence of the paired fastq file.
Look at what the `check_samplesheet.py` script is doing there

File: backup/nextflow.config [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  // Defaults only, expecting to be overwritten
+  max_memory = 128.GB
+  max_cpus = 16
+  max_time = 240.h
+
+}
+
+// Container slug. Stable releases should specify release tag!
+// Developmental code should specify :dev
+process.container = 'nfcore/rnavar:dev'
```
Comment (maxulysse) [inline backup/nextflow.config:51]:
```suggestion
```
We won't need a main container for this pipeline as we're going to have a container per process approach with DSL2

File: conf/base.config [hunk 3]
```diff
@@ -1,51 +1,50 @@
...
- *  nf-core/rnavar Nextflow base config file
- * -------------------------------------------------
- * A 'blank slate' config file, appropriate for general
- * use on most high performace compute environments.
- * Assumes that all software is installed and available
- * on the PATH. Runs in `local` mode - all jobs will be
- * run on the logged in environment.
- */
+========================================================================================
+    nf-core/rnaseq Nextflow base config file
```
Comment (maxulysse) [inline conf/base.config:3]:
```suggestion
    nf-core/rnavar Nextflow base config file
```

File: conf/modules.config [hunk 4]
```diff
@@ -0,0 +1,250 @@
+/*
+========================================================================================
+    Config file for defining DSL2 per module options
```
Comment (maxulysse) [inline conf/modules.config:3]: we should only have config for the tools we actually have here

File: conf/test.config [hunk 5]
```diff
@@ -1,28 +1,47 @@
...
+    config_profile_name        = 'Test profile'
+    config_profile_description = 'Minimal test dataset to check pipeline function'
+
+    // Limit resources so that this can run CI
+    max_cpus   = 2
+    max_memory = 6.GB
+    max_time   = 6.h
+
+    // Input data
+    input = 'https://raw.githubusercontent.com/nf-core/test-datasets/rnaseq/samplesheet/v3.1/samplesheet_test.csv'
```
Comment (maxulysse) [inline conf/test.config:23]: Make sure that the input for the test matches what we have

File: lib/WorkflowMain.groovy [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+class WorkflowMain {
+
+    //
+    // Citation string for pipeline
+    //
+    public static String citation(workflow) {
+        return "If you use ${workflow.manifest.name} for your analysis please cite:\n\n" +
+            "* The pipeline\n" +
+            "  https://doi.org/10.5281/zenodo.1400710\n\n" +
```
Comment (maxulysse) [inline lib/WorkflowMain.groovy:13]: We shouldn't have a zenodo doi for this pipeline yet

File: lib/WorkflowRnaseq.groovy [hunk 7]
```diff
@@ -0,0 +1,245 @@
+//
+// This file holds several functions specific to the workflow/rnaseq.nf in the nf-core/rnaseq pipeline
+//
+
+class WorkflowRnaseq {
```
Comment (maxulysse) [inline lib/WorkflowRnaseq.groovy:5]:
```suggestion
class WorkflowRnavar {
```
file should be renamed as well

**Comment by maxulysse** [comment]
Not sure about the current samplesheet.
But you seemed to have a good start.
There is still a lot from the rnaseq pipeline that we might not need here.
Can you try to fix the tests as well?

File: conf/test.config (hunk 5 above)
Comment (praveenraj2018) [reply conf/test.config:23]: yes it matched. I have not included single_end flag

File: lib/WorkflowMain.groovy (hunk 6 above)
Comment (praveenraj2018) [reply lib/WorkflowMain.groovy:13]: Yes, I just commented it and mention as "unpublished"

File: conf/modules.config (hunk 4 above)
Comment (praveenraj2018) [reply conf/modules.config:3]: changed

File: conf/base.config (hunk 3 above)
Comment (praveenraj2018) [reply conf/base.config]: changed

**Comment by maxulysse** [comment]
OK, so CI is now working, and test is complete.
That's very good.
I think we can merge that as it is for now.

**Review by maxulysse — APPROVED** [review]

**Pull request merged**